### PR TITLE
UNOMI-371 add optional support for optimistic concurrency control (if_seq_no)

### PR DESCRIPTION
**Pull request by lyogev**
When profiles are written at the same time (multiple events on the same profile for example, or segment updates during other segment updates) sometimes we are getting inconsistencies with the final profile.
This PR adds an optional flag to start using elasticsearch optimistic concurrency mechanism
This, in conjunction with the throwExceptions flag allows the client to be aware of such a case and retry if needed.

### CODE REVIEW

**Comment by lyogev** [comment]
@sergehuber I'm thinking that maybe to make this a more robust change we need to get the PersistenceService to return the new saved object whenever a save is triggered (changing the signature for all the save functions), WDYT?

**Review by sergehuber — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm requesting a change on the name of a field because I'm afraid it's going to add confusion. Unfortunately I don't have a better name to provide at this point. Options maybe ?

**Comment by sergehuber** [comment]
I just realized I forgot to answer your question about the save methods. That's ok with me as long as we don't change any REST or GraphQL API endpoints.

File: api/src/main/java/org/apache/unomi/api/Item.java
Comment (sergehuber) [reply api/src/main/java/org/apache/unomi/api/Item.java:67]: @lyogev have you seen my previous message about the metadata name ? Once this is fixed we could merge this.

Comment (lyogev) [reply api/src/main/java/org/apache/unomi/api/Item.java:67]: Hi @sergehuber yes I've seen the comment and I will update, but we are adding more code here soon as we are already using this mode, and we discovered some more issues. will update soon

Comment (sergehuber) [reply api/src/main/java/org/apache/unomi/api/Item.java:67]: Ok thanks for the feedback. Will probably merge this after 1.5.3 then.

**Comment by sergehuber** [comment]
Hello @lyogev , due to other merges this PR now conflicts with the master. Could you fix it ?

Also for the metadata I could suggest we rename it to systemMetadata ?

Thanks for your help and contribution. 

Regards,
  Serge...

**Comment by giladw** [comment]
Hi @sergehuber ,
im opening a new PR with the requested changes. and closing this one.
i hope its ok :)

**Comment by lyogev** [comment]
Closing this PR and moving to #223

**Pull request closed without merging**
